File: Dapper.WebApi/Models/ProductImage.cs
using DapperWebApi.Models;

namespace Dapper.WebApi.Models
{
    public class ProductImage
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public string FileName { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public bool IsMain { get; set; } = false;
        public int SortOrder { get; set; } = 0;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: react_project/src/types/ProductImage.ts
export interface ProductImage {
    id: number;
    productId: number;
    fileName: string;
    url: string;
    isMain: boolean;
    sortOrder: number;
    createdAt: string;
}

export interface ProductWithImages {
    id: number;
    name: string;
    description: string;
    price: number;
    createdAt?: string;
    mainImage?: ProductImage;
    images?: ProductImage[];
}